code/modules/mob/living/simple_animal/friendly/mule_bot/mule_bot.dm

```diff
@@ -0,0 +1,202 @@
+			span_warning("You patch some dents on \the [src]."))
+		if(src.heal_limb_damage(15,15, robo_repair = FALSE, updating_health = TRUE))
+			UpdateDamageIcon()
+		if(!I.tool_use_check(user, 2))
+			break
+		if(!src.bruteloss)
+			balloon_alert(user, "Dents fully repaired.")
+			break
+	cut_overlay(GLOB.welding_sparks)
+	return TRUE
+
+/mob/living/simple_animal/mule_bot/crowbar_act(mob/living/user, obj/item/I)
+	remove_module()
+	return TRUE
+
+
+/mob/living/simple_animal/mule_bot/proc/swap_module(obj/item/mule_module/mod, mob/user)
+	remove_module(user, FALSE)
+	apply_module(mod,user, FALSE)
+	update_icon()
+
+/mob/living/simple_animal/mule_bot/proc/apply_module(obj/item/mule_module/mod, mob/user, update_icon = TRUE)
+	if(mod.apply(src))
+		to_chat(user,span_notice("You succesfully installed [mod]"))
+		user?.temporarilyRemoveItemFromInventory(mod)
+	else
+		to_chat(user,span_warning("You failed to installed [mod]"))
+	if(update_icon)
+		update_icon()
+
+/mob/living/simple_animal/mule_bot/proc/remove_module(mob/user, update_icon = TRUE)
+	if(!installed_module)
+		return
+	if(user)
+		to_chat(user,span_notice("You succesfully remove [installed_module]"))
+	installed_module.unapply(FALSE)
+	if(update_icon)
+		update_icon()
+
+
+/obj/item/remote
+	name = "remote"
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "multitool2"
+	var/mob/living/simple_animal/mule_bot/bot
+
+/obj/item/remote/attack_self(mob/user)
+	if(!bot)
+		return
+	if(SEND_SIGNAL(src, COMSIG_REMOTECONTROLL_STOP_FOLLOW, usr) & COMSIG_BOT_STOP)
+		to_chat(user,span_notice("[bot] will now stop following"))
+	else
+		to_chat(user,span_notice("[bot] wil now follow you"))
+
+
+/obj/item/remote/afterattack(turf/T, mob/living/user)
+	. = ..()
+	SEND_SIGNAL(src, COMSIG_SET_TARGET, user, T)
+	to_chat(user, span_notice("[user] is now following [T]"))
+
+
+/*
+The bot has 2 modes
+Follow and goto
+
+In follow, it will follow the remote and who ever is holding it
+in goto. it will go to a selected tile and stay there
+
+*/
+/datum/ai_behavior/mule_bot
+	target_distance = 1
+	base_action = ESCORTING_ATOM
+	//The atom that will be used in only_set_escorted_atom proc, by default this atom is the remote
+	var/datum/weakref/default_escorted_atom
+	var/follow = FALSE
+
+/datum/ai_behavior/mule_bot/New(loc, parent_to_assign, escorted_atom, can_heal = FALSE)
+	. = ..()
+	default_escorted_atom = WEAKREF(escorted_atom)
+	RegisterSignal(escorted_atom, COMSIG_REMOTECONTROLL_STOP_FOLLOW, PROC_REF(stop_follow))
+	RegisterSignal(escorted_atom, COMSIG_SET_TARGET, PROC_REF(go_to_obj_target))
+
+/datum/ai_behavior/mule_bot/proc/stop_follow(atom/source, mob/user)
+	SIGNAL_HANDLER
+	if(follow)
+		follow = FALSE
+		atom_to_walk_to = null
+		change_action(MOVING_TO_ATOM)
+		mob_parent.say("Ill wait right here!")
+		return COMSIG_BOT_STOP
+	else
+		follow = TRUE
+		atom_to_walk_to = source
+		change_action(ESCORTING_ATOM, source)
+		mob_parent.say("Wait im up coming!")
+		return COMSIG_BOT_FOLLOW
```

Comment:
```suggestion
	follow = TRUE
	atom_to_walk_to = source
	change_action(ESCORTING_ATOM, source)
	mob_parent.say("Wait im up coming!")
	return COMSIG_BOT_FOLLOW
```
Sidenote review your grammer in your say blurbs

Reply:
I could entirly remove that toggeling the remote stops the bot since you can just click on the tile where you want it to stop. And free's up that control for other stuff in the future